src/converse-chatview.js

```diff
@@ -344,11 +344,19 @@
                                "Output has been shortened."),
                             true, true);
                     }
+
+                    var mess_time;
+
+                    if(_converse.time_format === 'h')
+                        mess_time = msg_time.format('hh:mm a');
+                    else if(_converse.time_format === 'H')
+                        mess_time = msg_time.format('HH:MM');
+
                     var $msg = $(template(
                         _.extend(this.getExtraMessageTemplateAttributes(attrs), {
                             'msgid': attrs.msgid,
                             'sender': attrs.sender,
-                            'time': msg_time.format('hh:mm'),
+                            'time': mess_time,
```

Comment:
No. That's not what I meant. We want the user to specify the format. Whatever format that is. Don't try to restrict the format that the user can provide.